lib/ansible/module_utils/network/nxos/nxos.py

```diff
@@ -951,69 +1019,55 @@ def get_proposed(self):
         ref = self._ref
         # '_proposed' may be empty list or contain initializations; e.g. ['feature foo']
         proposed = ref['_proposed']
+
+        if 'absent' in ref['_state'] and ref['_context'] and ref['_context'][-1].startswith('no'):
+            [proposed.append(ctx) for ctx in ref['_context']]
+            return proposed
+
         # Create a list of commands that have playbook values
-        play_keys = [k for k in ref['commands'] if 'playval' in ref[k]]
+        play_keys = ref['_playkeys'] = [k for k in ref['commands'] if 'playval' in ref[k]]
+
+        def have_equal_want(playval, existing):
+            if 'present' in ref['_state']:
+                if existing is None:
+                    return False
+                elif playval == existing or playval in existing.values():
+                    return True
+
+            if 'absent' in ref['_state']:
+                if isinstance(existing, dict) and all(x is None for x in existing.values()):
+                    existing = None
+                if existing is None or playval not in existing.values():
+                    return True
+            return False
 
         # Compare against current state
         for k in play_keys:
             playval = ref[k]['playval']
             existing = ref[k].get('existing', ref[k]['default'])
-            if playval == existing and ref['_state'] == 'present':
-                continue
-            if isinstance(existing, dict) and all(x is None for x in existing.values()):
-                existing = None
-            if existing is None and ref['_state'] == 'absent':
-                continue
-            cmd = None
-            kind = ref[k]['kind']
-            if 'int' == kind:
-                cmd = ref[k]['setval'].format(playval)
-            elif 'list' == kind:
-                cmd = ref[k]['setval'].format(*(playval))
-            elif 'dict' == kind:
-                # The setval pattern should contain placeholder keys that
-                # match up with the getval regex named group keys; e.g.
-                #   getval: my-cmd (?P<foo>\d+) bar (?P<baz>\d+)
-                #   setval: my-cmd {foo} bar {baz}
-                cmd = ref[k]['setval'].format(**playval)
-            elif 'str' == kind:
-                if 'deleted' in playval:
-                    if existing:
-                        cmd = 'no ' + ref[k]['setval'].format(existing)
-                else:
-                    cmd = ref[k]['setval'].format(playval)
+
+            # Multiple Instances:
```

Comment:
This section is confusing to me. Might need a codewalk...

Seems like you should be able to move the separate `Multiple Instances:` sections into the same section?

Also, we wanted to make *all* of the CmdRef `getval`/`setval` patterns use named groups. If we did that we can streamline this since everything will be a dict?

Reply:
We chatted about this.  I tried your suggestion of a for/else block but could not get it to work quite right.  What I have now works so I would prefer to handle this in a future commit.